addons/web/static/src/core/utils/objects.js

```diff
@@ -27,6 +27,24 @@ export function deepCopy(obj) {
     return JSON.parse(JSON.stringify(obj));
 }
 
+/**
+ * Check if an object is an instance of the 'type' interface,
+ * regardless of the window it belongs to.
+ * This is useful i.e. when the object to check is inside an iframe.
+ *
+ * @param {*} obj
+ * @param {Function} type
+ * @returns true if obj is an instance of type, regardless of the window it belongs to
+ */
+export function isInstanceOf(obj, type) {
+    for (let p = obj.__proto__; p; p = p.__proto__) {
```

Comment:
Don't use `__proto__`, it's deprecated and can be shadowed by `Object.defineProperty`, prefer `Object.<get/set>PrototypeOf`

Reply:
Thanks Sam, finally we will not introduce this function anyway ;-)